daml-lf/spec/daml-lf-1.rst

```diff
@@ -2569,6 +2575,150 @@ Map functions
 
   [*Available in versions >= 1.3*]
 
+
+Generic Map functions
+~~~~~~~~~~~~~~~~~~~~~
+
+* ``GENMAP_EMPTY : ∀ α. ∀ β. 'GenMap' α β``
+
+  Returns the empty generic map.
+
+  [*Available in versions >= 1.dev*]
+
+* ``GENMAP_INSERT : ∀ α. ∀ β.  α → β → 'GenMap' α β → 'GenMap' α β``
+
+  Inserts a new key and value in the map. If the key is already
+  present in the map, the associated value is replaced with the
+  supplied value.
+
+  This raises an error if the key is not a valid map key. Keys are
+  compared according to the rules listed below.
+
+  [*Available in versions >= 1.dev*]
+
+* ``GENMAP_LOOKUP : ∀ α. ∀ β.  α → 'GenMap' α β → 'Optional' α``
+
+  Lookups the value at a key in the map.
+
+  This raises an error if the key is not a valid map key. Keys are
+  compared according to the rules listed below.
+
+  [*Available in versions >= 1.dev*]
+
+* ``GENMAP_DELETE : ∀ α. ∀ β.  α → 'GenMap' α β → 'GenMap' α β``
+
+  Deletes a key and its value from the map. When the key is not a
+  member of the map, the original map is returned.
+
+  This raises an error if the key is not a valid map key. Keys are
+  compared according to the rules listed below.
+
+  [*Available in versions >= 1.dev*]
+
+* ``GENMAP_LIST : ∀ α. ∀ β.  'GenMap' α β → 'List' ⟨ key: α, value: β ⟩``
+
+  Converts to a list of key/value pairs. Unlike ``'MAP_LIST'``, the output
+  list is guaranteed to be in the order that the keys were first inserted.
+
+  [*Available in versions >= 1.dev*]
+
+* ``GENMAP_SIZE : ∀ α. ∀ β.  'GenMap' α β → 'Int64'``
+
+  Return the number of elements in the map.
+
+  [*Available in versions >= 1.dev*]
+
+
+**Validity of Keys:** A key is valid if and only if it is equivalent
+to itself according to the relation ``~ᵥ`` defined below. Attempts to
+use an invalid key in the operations above always result in a runtime
+error.
+
+Of particular note, the following values are never valid keys:
+
+* Lambda expressions ``λ x : τ . e``
+* Type abstractions ``Λ α : k . e``
+* (Partially applied) built-in functions
+* (Un-evaluated) Update/Scenario values
+* Any value containing an invalid key
+
+**Comparison of Keys:** We define here the relation ``~ᵥ`` on value
+expressions that is used for key comparison. This is a partial
+equivalence relation over all values, but a (total) equivalence
+relation over valid keys.
+
+This relation is not exposed as a builtin function, but it coincides
```

Comment:
Currently you are switching between `builtin` and `built-in`, it would probably be good to stick to one of those.